### PR TITLE
Map temperature values of >255 to None

### PR DESCRIPTION
**Pull request by jkeljo**
I got a bug report about sudden and temporary spikes in temperature values: https://github.com/jkeljo/hacs-greeneye-monitor/issues/35

Looking at [btmon.py](https://github.com/matthewwall/mtools/blob/master/bin/btmon.py#L1897), temperature values above 255 indicate that it was unable to read the sensor. Also the DS18B20 sensor that the GEM expects is [documented](https://www.analog.com/media/en/technical-documentation/data-sheets/DS18B20.pdf) to have a range up to 125C/257F, with signed 9-bit values being used to report 0.5C increments, so the max temp it could report is 127.5C/261.5F. Given that the bug report is from someone with their GEM set to Fahrenheit, it looks like the GEM is doing some processing of the data. Either way, 255 is pretty close to the max you'd ever see out of the sensor.

My own GEM, which has no temperature sensors connected, reports 256 for some channels and 0 for others, so it's not super consistent. However, it seems pretty clear that values above 255 indicate bad data.

### CODE REVIEW

**Comment by jkeljo** [comment]
Note that this is a breaking change because now values can be `None` that couldn't be before. If you'd rather not take a major version bump for this, I'm fine with fixing it further upstack.

File: tests/gem/packet_test_data.py [hunk 1]
```diff
@@ -82,7 +82,7 @@
         "seconds": 997492,
         "serial_number": 603,
         "device_id": 11,
-        "temperatures": [256, -5, 0, 0, 0, 0, 0, 0],
```
Comment (sdwilsh) [inline tests/gem/packet_test_data.py:85]: The GitHub UI makes this a little hard to see in the PR view; do we have any tests with a valid temp?

**Comment by sdwilsh** [comment]
Let's plan to make this fix part of 1.0 I guess.  It's a major bump, but this is maybe the last big change we need to worry about?

File: tests/gem/packet_test_data.py (hunk 1 above)
Comment (jkeljo) [reply tests/gem/packet_test_data.py:85]: -5 is a valid temperature :)

Comment (sdwilsh) [reply tests/gem/packet_test_data.py:85]: COVID shot gave me a fever.  Clearly I shouldn't have been reviewing code...

Comment (sdwilsh) [reply tests/gem/packet_test_data.py:85]: Can we get at least one positive value in these tests as well so we cover all of our bases?

**Comment by Vlad-Star** [comment]
Hey guys! I am the original author of [35](https://github.com/jkeljo/hacs-greeneye-monitor/issues/35), just got back from vacation. Is there something I can help with to move forward with this PR? @sdwilsh @jkeljo

**Comment by sdwilsh** [comment]
@Vlad-Star, I can approve and merge what @jkeljo has here if you want to update the tests in a new PR.  Although, that won't help get the custom integration updated, so hopefully we can hear back from @jkeljo soon!

**Comment by jkeljo** [comment]
Sorry guys, I've had some pretty busy weekends lately. Hopefully I'll have time to break out the hex editor and make those packets have some positive values soon.

**Pull request merged**